waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/FederatedHMSHandler.java

```diff
@@ -1157,10 +1157,18 @@ public GetRoleGrantsForPrincipalResponse get_role_grants_for_principal(GetRoleGr
   @Loggable(value = Loggable.DEBUG, skipResult = true, name = INVOCATION_LOG_NAME)
   public PrincipalPrivilegeSet get_privilege_set(HiveObjectRef hiveObject, String user_name, List<String> group_names)
       throws MetaException, TException {
-    DatabaseMapping mapping = databaseMappingService.databaseMapping(hiveObject.getDbName());
-    return mapping
-        .getClient()
-        .get_privilege_set(mapping.transformInboundHiveObjectRef(hiveObject), user_name, group_names);
+    DatabaseMapping mapping;
+    HiveObjectRef hiveObjectResult;
+    if (hiveObject.getDbName() == null) {
+      mapping = databaseMappingService.primaryDatabaseMapping();
+      hiveObjectResult = hiveObject;
```

Comment:
you'll need to transform the object for this mapping as well (it works like this but will be more future proof if we transform, probably easiest to remove the `hiveObjectResult` field and change the return to the original:
```
mapping.getClient().get_privilege_set(mapping.transformInboundHiveObjectRef(hiveObject), user_name, group_names);
```

Reply:
I have change the code like this 

public PrincipalPrivilegeSet get_privilege_set(HiveObjectRef hiveObject, String user_name, List<String> group_names)
      throws MetaException, TException {
    DatabaseMapping mapping;
    if (hiveObject.getDbName() == null) {
      mapping = databaseMappingService.primaryDatabaseMapping();
    } else {
      mapping = databaseMappingService.databaseMapping(hiveObject.getDbName());
    }
    return mapping.getClient().get_privilege_set(mapping.transformInboundHiveObjectRef(hiveObject), user_name,
            group_names);
  }